projects/igniteui-angular/src/lib/date-common/util/date-time.util.ts

```diff
@@ -0,0 +1,525 @@
+            newDate.setDate(maxDate);
+        }
+        newDate.setFullYear(newDate.getFullYear() + delta);
+    }
+
+    /** Spins the hours portion in a date-time editor. */
+    public static spinHours(delta: number, newDate: Date, isSpinLoop: boolean): void {
+        const maxHour = 23;
+        const minHour = 0;
+        let hours = newDate.getHours() + delta;
+        if (hours > maxHour) {
+            hours = isSpinLoop ? hours % maxHour - 1 : maxHour;
+        } else if (hours < minHour) {
+            hours = isSpinLoop ? maxHour + (hours % maxHour) + 1 : minHour;
+        }
+
+        newDate.setHours(hours);
+    }
+
+    /** Spins the minutes portion in a date-time editor. */
+    public static spinMinutes(delta: number, newDate: Date, isSpinLoop: boolean): void {
+        const maxMinutes = 59;
+        const minMinutes = 0;
+        let minutes = newDate.getMinutes() + delta;
+        if (minutes > maxMinutes) {
+            minutes = isSpinLoop ? minutes % maxMinutes - 1 : maxMinutes;
+        } else if (minutes < minMinutes) {
+            minutes = isSpinLoop ? maxMinutes + (minutes % maxMinutes) + 1 : minMinutes;
+        }
+
+        newDate.setMinutes(minutes);
+    }
+
+    /** Spins the seconds portion in a date-time editor. */
+    public static spinSeconds(delta: number, newDate: Date, isSpinLoop: boolean): void {
+        const maxSeconds = 59;
+        const minSeconds = 0;
+        let seconds = newDate.getSeconds() + delta;
+        if (seconds > maxSeconds) {
+            seconds = isSpinLoop ? seconds % maxSeconds - 1 : maxSeconds;
+        } else if (seconds < minSeconds) {
+            seconds = isSpinLoop ? maxSeconds + (seconds % maxSeconds) + 1 : minSeconds;
+        }
+
+        newDate.setSeconds(seconds);
+    }
+
+    /** Spins the AM/PM portion in a date-time editor. */
+    public static spinAmPm(newDate: Date, currentDate: Date, amPmFromMask: string): Date {
+        switch (amPmFromMask) {
+            case 'AM':
+                newDate = new Date(newDate.setHours(newDate.getHours() + 12));
+                break;
+            case 'PM':
+                newDate = new Date(newDate.setHours(newDate.getHours() - 12));
+                break;
+        }
+        if (newDate.getDate() !== currentDate.getDate()) {
+            return currentDate;
+        }
+
+        return newDate;
+    }
+
+    /**
+     * Determines whether the provided value is greater than the provided max value.
+     *
+     * @param includeTime set to false if you want to exclude time portion of the two dates
+     * @param includeDate set to false if you want to exclude the date portion of the two dates
+     * @returns true if provided value is greater than provided maxValue
+     */
+    public static greaterThanMaxValue(value: Date, maxValue: Date, includeTime = true, includeDate = true): boolean {
+        if (!DateTimeUtil.isValidDate(value) || !DateTimeUtil.isValidDate(maxValue)) {
```

Comment:
Yeah this shouldn't really happen - there's no correct answer if the values to compare are not available.